ql/models/shortrate/onefactormodels/gsr.cpp

```diff
@@ -92,6 +92,12 @@ Gsr::Gsr(const Handle<YieldTermStructure> &termStructure,
     initialize(T);
 }
 
+void Gsr::update() { 
```

Comment:
Mostly related to the global optimizers. 
The changes to model.hpp and model.cpp are part of the SWIG pull-request, and they are intended to expose the state of the model, so that its performance could be better evaluated. 
In markovfunctional.hpp, I just added a constructor that allowed to set all the model settings at once, instead of having to start with default values and then call the setter methods for each one. 

However, the changes to Gsr are simply a very small refactoring, which I did while reviewing Gsr for the global optimizer. 
For Gsr::update(), I moved the flushCache precondition that updateTimes and updateReversion had to call to inside of the update function. 
For Gsr::initialize(), I moved the reversion loop inside the else case where it would be needed. In the other case, it is a constant, so there was no need to loop through  it. And then, since I was moving that, I moved the sigma initialization to just before the loop, so that they would be together.